demo/demo.sh

```diff
@@ -14,14 +14,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cd $(dirname ${BASH_SOURCE})
-. ./udemo.sh
-. ../scripts/util.sh
+# Resolve the absolute path to this script directory.
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Source the helper scripts with absolute paths.
+. "${script_dir}/udemo.sh"
+. "${script_dir}/../scripts/util.sh"
 
 DEMO_AUTO_RUN=true
 
-kubeconfig1=${KUBECONFIG1:-../scripts/c1.kubeconfig}
-kubeconfig2=${KUBECONFIG2:-../scripts/c2.kubeconfig}
+# Use absolute paths for kubeconfig defaults.
+kubeconfig1="${KUBECONFIG1:-$(realpath "${script_dir}/../scripts/c1.kubeconfig")}"
+kubeconfig2="${KUBECONFIG2:-$(realpath "${script_dir}/../scripts/c2.kubeconfig")}"
```

Comment:
This is already an absolute path, realtpath looks redundant here or am I missing something?

Reply:
Thank you for your feedback. I used `realpath` to prevent paths like `/hoge/fuga/../scripts/c1.kubeconfig` from appearing when using `echo`, aiming for a cleaner output such as `/hoge/scripts/c1.kubeconfig`. However, as you pointed out, it doesn't have any functional impact and is merely a matter of personal preference. I'm not particularly attached to it, so removing `realpath` should be perfectly fine. What are your thoughts on this?

Reply:
Ah yes ok, thanks for the explanation ,no worries then!